runtime-testsuite/test/org/antlr/v4/test/runtime/BaseRuntimeTest.java

```diff
@@ -62,7 +62,7 @@ public static void startHeartbeatToAvoidTimeout() {
 	}
 
 	private static boolean requiresHeartbeat() {
-		return isTravisCI() || isAppVeyorCI() || (isCPP() && isRecursion()) || (isCircleCI() && isGo());
+		return isTravisCI() || isAppVeyorCI() || (isCPP() && isRecursion()) || (isCircleCI() && isGo())|| (isCircleCI() && isCSharp() && isRecursion());
```

Comment:
Minor: `&&` has higher priority than `||`. Thus, parentheses are excess:

Also, it looks more clear if it's splitted on several lines:

```suggestion
		return isTravisCI() ||
		       isAppVeyorCI() ||
		       isCPP() && isRecursion() ||
		       isCircleCI() && isGo() ||
		       isCircleCI() && isCSharp() && isRecursion();
```